db/background_mgr_resync_dcp.go

```diff
@@ -85,29 +74,34 @@ func (r *ResyncManagerDCP) Init(ctx context.Context, options map[string]interfac
 	// add collection list to manager for use in status call
 	r.SetCollectionStatus(collectionNames)
 
-	if clusterStatus != nil {
-		var statusDoc ResyncManagerStatusDocDCP
-		err := base.JSONUnmarshal(clusterStatus, &statusDoc)
-
-		reset, ok := options["reset"].(bool)
-		if reset && ok {
-			base.InfofCtx(ctx, base.KeyAll, "Resync: Resetting resync process. Will not resume any partially completed process")
-		}
-
-		// If the previous run completed, or there was an error during unmarshalling the status we will start the
-		// process from scratch with a new resync ID. Otherwise, we should resume with the resync ID, stats specified in the doc.
-		if statusDoc.State == BackgroundProcessStateCompleted || err != nil || (reset && ok) {
-			return newRunInit()
+	// If the previous run completed, or there was an error during unmarshalling the status we will start the
+	// process from scratch with a new resync ID. Otherwise, we should resume with the resync ID, stats specified in the doc.
+	var resetReason string
+	var statusDoc ResyncManagerStatusDocDCP
+	if clusterStatus == nil {
+		resetReason = "no previous run found"
+	} else if resetOpt, _ := options["reset"].(bool); resetOpt {
+		resetReason = "reset option requested"
+	} else if err := base.JSONUnmarshal(clusterStatus, &statusDoc); err != nil {
+		resetReason = "failed to unmarshal cluster status"
+	} else if statusDoc.State == BackgroundProcessStateCompleted {
+		resetReason = "previous run completed"
+	}
+	if resetReason != "" {
+		newID, err := uuid.NewRandom()
+		if err != nil {
+			return err
 		}
-		r.ResyncID = statusDoc.ResyncID
-		r.SetStatus(statusDoc.DocsChanged, statusDoc.DocsProcessed)
-
-		base.InfofCtx(ctx, base.KeyAll, "Resync: Attempting to resume resync with resync ID: %s", r.ResyncID)
-
+		r.ResyncID = newID.String()
+		base.InfofCtx(ctx, base.KeyAll, "Resync: Resetting resync process with ID: %q - %s", r.ResyncID, resetReason)
```

Comment:
I am fine with this change but I wonder if the two common scenarios here actually should not have the log line reset? I am not convinced by this, but I worry reset implies dropping previous bad state.

`clusterStatus == nil` - technically this is a reset because you are "setting" but I think that this is better stated as just "Starting resync process".

Similarly with `statusDoc.State == BackgroundProcessStateCompleted` but this case I can see how reset seems like the right word.